src/QMCHamiltonians/LatticeDeviationEstimator.cpp

```diff
@@ -0,0 +1,238 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Yubo Yang, paul.young.0414@gmail.com, University of Illinois at Urbana-Champaign
+//
+// File created by: Yubo Yang, paul.young.0414@gmail.com, University of Illinois at Urbana-Champaign
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "QMCHamiltonians/LatticeDeviationEstimator.h"
+#include <OhmmsData/AttributeSet.h>
+
+namespace qmcplusplus
+{
+
+LatticeDeviationEstimator::LatticeDeviationEstimator(ParticleSet& P,ParticleSet& sP, std::string tgroup_in, std::string sgroup_in): 
```

Comment:
Unless there's a good reason to do otherwise, strings should be passed as a reference and made const
(const std::string &tgroup_in)
The 'const' makes it possible to use a literal character string when calling (useful for testing)
Passing by reference avoids extra copies.